plugin-api/src/main/java/org/sonar/api/rules/RuleCharacteristic.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Sonar Plugin API
+ * Copyright (C) 2009-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.api.rules;
+
+/**
+ * Enum that represents Characteristic according to the new Clean Code Taxonomy. Should be used instead of {@link RuleType} as it will
+ * become mandatory in the future to provide it.
+ *
+ * @since 9.16
+ */
+public enum RuleCharacteristic {
+
+  // Fit for development
+  CLEAR(1), CONSISTENT(2), STRUCTURED(3), TESTED(4),
+  // Fit for production
+  ROBUST(5), SECURE(6), PORTABLE(7), COMPLIANT(8);
+
+  private final int dbConstant;
```

Comment:
I guess pros and cons have already been debated, but just in case: should we really expose such internal concepts in the API?

Reply:
As the concept of characteristics will evolve, we may have to deprecate some values and introduce new ones. Maybe just for renaming. We could end up with multiple enum entries having the same dbConstant.

Reply:
To me it makes sense to drop them. It is copy-paste from RuleType enum and I am not sure how the discussion went for RuleType in the past, I assumed there was a strong case for them for RuleType. 

I will remove them and if someone has a strong argument to keep them in plugin-api feel free to re-open this discussion.